### PR TITLE
[mono][aot] Make the dedup mode output deterministic.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: src/mono/mono/mini/aot-compiler.c [hunk 1]
```diff
@@ -448,6 +448,7 @@ static MonoAotCompile *llvm_acfg;
 static MonoAotCompile *current_acfg;
 static MonoAssembly *dedup_assembly;
 static GHashTable *dedup_methods;
+static GPtrArray *dedup_methods_list;
```
Comment (fanyang-mono) [inline src/mono/mono/mini/aot-compiler.c:451]: How do you plan to use the data saved in this array? Do you want to print it out in certain verbose mode or output it in a file by setting a switch?

**Review by fanyang-mono — APPROVED** [review]
LGTM!

File: src/mono/mono/mini/aot-compiler.c (hunk 1 above)
Comment (vargaz) [reply src/mono/mono/mini/aot-compiler.c:451]: Its just used to iterate over the methods in a fixed order. The hashtable iteration order depends on the hash codes of MonoMethod*'s which are not stable.

**Pull request merged**